### PR TITLE
Gen4 refactoring and clean up

### PR DESCRIPTION
**Pull request by systay**
## Description
Fixes an issue where the gen4 planner was not looking at the tables of columns when comparing them to other columns or with vindexes.

Also includes a non-trivial amount of clean ups.

## Related Issue(s)
#7280 

## Checklist
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vtgate/planbuilder/insert.go [hunk 1]
```diff
@@ -146,7 +146,7 @@ func buildInsertShardedPlan(ins *sqlparser.Insert, table *vindexes.Table) (engin
 	case sqlparser.Values:
 		rows = insertValues
 		if hasSubquery(rows) {
-			return nil, errors.New("unsupported: subquery in insert values")
+			return nil, errors.New("unsupported: simpleProjection in insert values")
```
Comment (GuptaManan100) [inline go/vt/vtgate/planbuilder/insert.go:149]: I don't think we should change this error message

**Review by GuptaManan100 — APPROVED** [review]
Everything else, LGTM 🚢  it

File: go/vt/vtgate/planbuilder/gen4_planner.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+	ksName := ""
+	if ks, _ := vschema.DefaultKeyspace(); ks != nil {
+		ksName = ks.Name
+	}
+	semTable, err := semantics.Analyze(sel, ksName, vschema, starRewrite)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx := newPlanningContext(reservedVars, semTable, vschema)
```
Comment (frouioui) [inline go/vt/vtgate/planbuilder/gen4_planner.go:76]: 👍🏻

**Review by frouioui — APPROVED** [review]
big refactor, that will be nice!

File: go/vt/vtgate/planbuilder/simple_projection.go [hunk 3]
```diff
@@ -23,31 +23,32 @@ import (
...
 // clause, because a route is more versatile than
-// a subquery.
-type subquery struct {
+// a simpleProjection.
+// this should not be used by the gen4 planner
+type simpleProjection struct {
 	logicalPlanCommon
 	resultColumns []*resultColumn
-	esubquery     *engine.Subquery
+	primitive     *engine.SimpleProjection
```
Comment (harshit-gangal) [inline go/vt/vtgate/planbuilder/simple_projection.go:41]: nit: in other logical plans, we use `e` as a suffix to indicate it is an engine.Primitive. We can follow the same suite here.

**Review by harshit-gangal — APPROVED** [review]

File: go/vt/vtgate/engine/simple_projection.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2019 The Vitess Authors.
```
[the remark this answers is not in the document]
Comment (systay) [reply go/vt/vtgate/engine/simple_projection.go:2]: I just renamed a file. git is confused and thinks it's a new file

**Pull request merged**